docs/StorageSequence.puml

```diff
@@ -0,0 +1,117 @@
+@startuml
+
+actor User
+participant ":Storage" as Storage
+participant ":File" as File
+participant ":FileWriter" as FileWriter
+participant ":Scanner" as Scanner
+participant ":ChessTile" as ChessTile
+participant ":String" as String
+
+
+create Storage
+User -> Storage++ : new Storage(): String
+create File
+Storage -> File++ : new File()
+File --> Storage-- : storageFile:File
+Storage --> User--
+
+
+User -> Storage++ : createChessMasterFile()
+Storage -> File++ : exists()
+File --> Storage-- : boolean
+alt File: !exists()
+    File -> File++ : createParentDirectories()
+    File --> File--
+    File -> File++ : createNewFile()
+    File --> File--
+end
+Storage --> User--
+
+
+User -> Storage++ : saveBoard(): ChessBoard
+Storage -> Storage++ : createChessMasterFile()
+Storage --> Storage--
+create FileWriter
+Storage -> FileWriter++ : new FileWriter(): File
```

Comment:
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/82555990/0e4ea6bb-065c-4bd7-a603-d9f1334bd388)
Is there supposed to be a lifeline for FileWriter?